Development v163rc1

sudwebdesign opened the pull request:
## Description
Same as #1267 BUT
- Adjust `setup` logic to add the `lower case` for all languages & the RC0 Change `include_zugferd` to `einvoice` (**Only `break`** the first loop after all steps reached). 
- And little Improvement of the `upload` controller of RC0 (_not reverted to old system_`*`)

## Motivation and Context
`*` It solve a problem when upload file with **accentuated** file
- Upload OK, Delete OK BUT when click on Download go to Red Screen Of Death (404)
- And after reload page the thumbnail not showed

## `*` Screenshots with #1267:
<details>
<summary>After upload (All seem fine)</summary>

![Screenshot 2025-05-20 at 14-44-54 InvoicePlane](https://github.com/user-attachments/assets/cd679459-86ca-4261-b85c-bc1b0ab257a8)

</details>
<details>
<summary>But after click on `Download` (404)</summary>

![Screenshot 2025-05-20 at 14-46-19 InvoicePlane - 404 Page Not Found](https://github.com/user-attachments/assets/19214eb9-bd87-405b-a562-6a0af8898a84)

</details>
<details>
<summary>And after return to invoice (missing thumb)</summary>

![Screenshot 2025-05-20 at 14-47-03 InvoicePlane](https://github.com/user-attachments/assets/f9b4c933-0422-486e-91e9-2609dd343f04)

</details>


## Pull Request Checklist

  * [x] My code follows the code formatting guidelines.
  * [ ] I have an issue ID for this pull request.
  * [x] I selected the corresponding branch.
  * [x] I have rebased my changes on top of the corresponding branch.

## Issue Type (Please check one or more)

  * [x] Bugfix of #1267 & #1237 PR
  * [ ] Improvement of an existing Feature
  * [ ] New Feature

Commit:
Improve versions in composer & package +
up yarn & composer lock

sudwebdesign committed:
Improve gh templates & workflows & infos (md)

Improve & Merge Develop v163rc1 (#1266)
Without TRANSLATION.md

sudwebdesign committed:
Setup: Upgrade default & users languages to lowercase like #1232

sudwebdesign committed:
Fix: Save products & tasks. No empty rules in Form_validation

Form_validation: set_rules() called with an empty $rules parameter

See: #1195

sudwebdesign committed:
Improve Uploader: Del old system. No show file \w upload_file()

Not used in v1.6.3RC0

sudwebdesign commented:
@nielsdrost7 

The [`upload controller`](https://github.com/InvoicePlane/InvoicePlane/blob/9553037a497c0dbfb51e5e9b2bc39622ef099cf2/application/modules/upload/controllers/Upload.php) work great in RC0. _have you tested before modifying file?_

It's inspired by [your work for v1.6.2](https://github.com/InvoicePlane/InvoicePlane/blob/1ba32beac378113ca285e5d54fb1404f985417fd/application/modules/upload/controllers/Upload.php) to make it (re)work (with numerous change like [this](https://github.com/InvoicePlane/InvoicePlane/blob/fa5a7acdabaffdda31d5a84885969305f3deb2c5/application/modules/upload/controllers/Upload.php) to use `db` & have an other logic of previous versions) 

Check the diff with v162 & v163RC0:
`git diff 1ba32beac378113ca285e5d54fb1404f985417fd 9553037a497c0dbfb51e5e9b2bc39622ef099cf2 -- application/modules/upload/controllers/Upload.php`

nielsdrost7 commented:
@sudwebdesign you've completely totally lost me.
We have a `develop-163rc1` branch and now also a `development-v163rc1` branch?

I thought the upload controller was broken and I tried to revert to the December 8th version.

I'm really lost.

I can be found on Discord of you want to explain.

By the way: you've got the rights to merge.
May I suggest to merge both branches into 1 branch and keep one of those branches in track with version 1.7.0?

Let me know when I need to make a new .zip file.

Our testers also need to know which branch to test

sudwebdesign commented:
@nielsdrost7 
Our testers use [`development-v163rc1`](https://github.com/InvoicePlane/InvoicePlane/tree/development-v163rc1)  branch to test the next RC1 (be careful with the `Folder` lang name: need to be go to lowercase (before use setup))
 
Just use this branch alias [`development-v163rc1`](https://github.com/InvoicePlane/InvoicePlane/tree/development-v163rc1) it's the same of [your (`md's`, gh workflow & templates) work in `develop-163rc1`](https://github.com/InvoicePlane/InvoicePlane/tree/develop-163rc1)  but the `upload` & `setup` work well here.

And Not use the next-1.7 branch [because of that](https://github.com/InvoicePlane/InvoicePlane/pull/1237/commits/b62d194ceb66f66f28121e89cc8a64f1de2d6e07) (same think of develop-163rc1). Set it to draft (or close it, & I resend another ASAP).

Sorry, I need to going out.

nielsdrost7 commented:
@sudwebdesign a suggestion to keep the branches `v163rc1` and `1.7.0` in line with each other.

commit https://github.com/InvoicePlane/InvoicePlane/commit/610ed0c8b91042b2f9cae135ecd7582627535dba from the 1.7.0 branch is missing in the `v163` branch (the empty rules) it can be cherry-picked from the `1.7.0` branch.

I won't touch either branches for a while so everything can be cleaned up and straightened out.

sudwebdesign committed:
PHP compat: No E_STRICT (error_reporting) & Adjust Rector rule

In accordance of doc, is unused & PHP 8.4 deprecate.
See:
https://www.php.net/manual/errorfunc.constants.php#constant.e-strict
---
Scope: production or testing environment
How to set? See:
https://github.com/orgs/InvoicePlane/discussions/1168

sudwebdesign committed:
Rector: More efficient Sets: deadCode, codeQuality & codingStyle

Applied rules:
 * RemoveUselessParamTagRector
 * StrictArraySearchRector
 * FuncGetArgsToVariadicParamRector

sudwebdesign committed:
Refacto \w Rector: Prepared set typeDeclarations:true

Applied rules:
 * ReturnUnionTypeRector
 * ReturnNullableTypeRector
 * RemoveUselessParamTagRector
 * RemoveUselessReturnTagRector
 * StrictStringParamConcatRector
 * StrictArrayParamDimFetchRector
 * SimplifyBoolIdenticalTrueRector
 * ParamTypeByMethodCallTypeRector
 * ReturnTypeFromStrictNewArrayRector
 * SimplifyEmptyCheckOnEmptyArrayRector
 * ReturnTypeFromReturnDirectArrayRector
 * NumericReturnTypeFromStrictReturnsRector
 * TypedPropertyFromStrictConstructorRector
 * BoolReturnTypeFromBooleanConstReturnsRector
 * BoolReturnTypeFromBooleanStrictReturnsRector
 * AddFunctionVoidReturnTypeWhereNoReturnRector
 * StringReturnTypeFromStrictStringReturnsRector

sudwebdesign committed:
Add composer scripts: phpcs, rector & check

`composer run check` to verify & correct the code (with all)

sudwebdesign committed:
Refacto: My_Form_validation::run() (Ready for Next-1.7)

Ok with CodeIgniter 3.1.13 (& 3.3 by pocketarc for PHP 8.2+)
See:
https://github.com/InvoicePlane/InvoicePlane/pull/1237/commits/e95b95f1888dbf0e4468a6aeabed73bd71b4360f

sudwebdesign commented:
Hi @nielsdrost7 IMHO it's ready to merge in development branch

**Be careful**
The obsolete #1237 renamed to [next-1.7-OLD](https://github.com/InvoicePlane/InvoicePlane/tree/next-1.7-OLD) & is OK to delete it (for me).

I remake the (**new**) [next-1.7](https://github.com/InvoicePlane/InvoicePlane/tree/next-1.7) & have only [1 commit to work with CI3.3](https://github.com/InvoicePlane/InvoicePlane/commit/e03200ee5bb1db14ed35a7ba935ea9e5fec2b02e) (in comparison of here)

nielsdrost7 commented:
@sudwebdesign merci, d'accord.

Do you want to build any other stuff into 1.7?
Maybe better handling of JavaScript errors?

We had that PHP notice couple of weeks ago which made JavaScript not respond correctly (my interpretation), the whole page got "stuck"

If I have time I'll make the .zip files for rc1 for both 1.6.3 and 1.7.0

sudwebdesign commented:
@nielsdrost7

> Do you want to build any other stuff into 1.7? 

_IMHO No_ or finish #1238 PDF Barcode 

> Maybe better handling of JavaScript errors? 

Like [this](https://github.com/orgs/InvoicePlane/projects/15/views/2?pane=issue&itemId=109216392) I've an Idea but for v1.6.3rc1, just wait ;)

sudwebdesign committed:
ipconfig: Add CI_ENV=production to hide minor PHP errors by default

Improve gitignore
& remove todo (oups)

sudwebdesign committed:
[wip] JSON.parse to json_parse to show error in front end

sudwebdesign committed:
[script.js] json_parse for all & console.trace(data) to debug

Need `yarn build`
if `ENABLE_DEBUG=false` in ipconfig

sudwebdesign committed:
[script.js] Finish json_parse: Add console.error & div.alert

sudwebdesign committed:
README header: Restore badges & show favicon at float right

sudwebdesign commented:
Hi @nielsdrost7

+ New `CI_ENV=production` in `ipconfig` (hide some PHP errors)
+ New system to show the data (toggled) on front-end when a parse json fail.

![Screenshot 2025-05-22 at 21-28-19 InvoicePlane](https://github.com/user-attachments/assets/c4cd157f-ecc5-4e11-ba7e-92c9cb254a58)
![Screenshot 2025-05-22 at 21-29-33 InvoicePlane](https://github.com/user-attachments/assets/68dd21a3-e2a4-49cb-a575-b217a3cb7d7a)

It's ready to test & (Maybe) OK to merge in development branch.

---

**Be careful** and _sorry_
I deleted the next-1.7 branch.
_BUT_
I create a _new_ [next-1.7.0](https://github.com/InvoicePlane/InvoicePlane/tree/next-1.7.0) & have only [1 commit to work with CI3.3](https://github.com/InvoicePlane/InvoicePlane/commit/254fe55ae8feeb0cdba502db3d0131b13778e3ce) (in [comparison of here](https://github.com/InvoicePlane/InvoicePlane/compare/development-v163rc1...next-1.7.0))

---

Note: I need a break.

nielsdrost7 commented:
> It's ready to test & (Maybe) OK to merge in development branch.

I'll make new .zip files.

> Note: I need a break.

Take your break man 👊

nielsdrost7 commented:
I'll make a new RC1 release today and then i'll post here.
@sudwebdesign when you're ready for the next branch, would you like to create that branch?
Also: can you merge "rc1" (which will be `development` after i merge this branch)
into version `next-1.7.0`?
